maestro-xcuitest-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -167,17 +167,29 @@ class XCTestDriverClient(
     }
 
     private fun OkHttpClient.Builder.addRetryOnErrorInterceptor() = addInterceptor(Interceptor {
+        // Kotlin lambda's can't contain while (true) loops..
+        retryOnError(it)
+    })
+
+    private fun retryOnError(it: Interceptor.Chain): Response {
+        var retryAttempt = 0
         val request = it.request()
-        try {
-            it.proceed(request)
-        } catch (connectException: IOException) {
-            if (installer.start()) {
-                it.proceed(request)
-            } else {
-                throw XCTestDriverUnreachable("Failed to reach out XCUITest Server in RetryOnError")
+        while (true) {
+            try {
+                return it.proceed(request)
+            } catch (connectException: IOException) {
+                if (retryAttempt >= retryCount) {
```

Comment:
Instead of having a `while(true)`. What do you think about having something like `while(retryAttempt <= retryCount)`?
This way we **guarantee** that we will never softlock maestro

Reply:
It would not execute the first proceed if retryCount == 0

Reply:
The retry logic should really be in the catch block

Reply:
My concern is only that for some unrelated reason the `proceed` fails with a different exception than IOException, which will not be captured by this catch block, resulting in an infinite loop.

Reply:
Ah, yes that case needs handling

Reply:
Actually, unhandled exceptions can never be trapped in the while loop. They would exit the loop immediately

Reply:
Oh yeah. You're right. It would throw the exception.